cfgov/v1/tests/test_links.py

```diff
@@ -22,6 +25,68 @@ def test_gov_link(self):
         self.assertIn('class="icon-link_text"', output)
 
 
+class LinkTest(TestCase):
+
+    def setUp(self):
+
+        self.page = mock.MagicMock()
+        self.request = mock.MagicMock()
+        self.request.url = 'http://localhost:8000/path/to/page/'
+        self.page.url = 'http://localhost:8000/path/to/some/other/page/'
+        self.page.live = True
+        self.page.specific.shared = True
+        self.context = mock.MagicMock()
+
+        def context_dict(name):
+            return {'request': self.request}[name]
+
+        self.context.__getitem__.side_effect = context_dict
+
+    def test_get_protected_url_no_page(self):
+        """
+        Confirm that in the absence of a page, get_protected_url returns the hash.
+        """
+        result = get_protected_url(self.context, None)
+        self.assertEqual(result, '#')
+
+    def test_get_protected_url_no_url(self):
+        """
+        Confirm that in the absence of a page URL, get_protected_url returns None.
+        """
+        self.page.url = None
+        result = get_protected_url(self.context, self.page)
+        self.assertIsNone(result)
+
+    def test_get_protected_url_live_page(self):
+        """
+        Confirm that if a live page is requested from the same host as the request
+        we get back the original page url.
+        """
+        self.page.url = 'http://localhost:8000/path/to/some/other/page/'
+        result = get_protected_url(self.context, self.page)
+        self.assertEqual(result, self.page.url)
+
+    def test_get_protected_url_non_live_page(self):
+        """
+        Confirm that if a non-live page is request, we get back the hash.
```

Comment:
Thanks for the tag @richaagarwal. Based on my initial understanding, I think this is the desired behavior but I don’t fully understand the test descriptions. Below are my not-a-developer questions, could you help me out @grapesmoker? 

> Confirm that in the absence of a page, get_protected_url returns the hash. 

I don’t know what “the absence of a page” might mean. Also not sure what returning “the hash” means as a behavior?

> Confirm that in the absence of a page URL, get_protected_url returns None.

Does this mean that any link would be null without a URL? If yes, +1

 > Confirm that if a live page is requested from the same host as the request we get back the original page url.

The reference to “original page url" throws me, but if this means that when a page is requested from www, we get back the requested page at www, then +1

> Confirm that if a non-live page is request, we get back the hash. aka, "Confirm that if a non-live page is requested from a live page, we get back the hash”

I think this is confirming that a staged page cannot be requested from www, which is good. Again, not sure what returning the hash means as a behavior.

> Make sure that if we're on the staging site and request a page from that site which does not point to the same host, that we replace the hostname with the staging hostname.

I think this means that if I’m on staging.cf.gov/about-us/ and I request www.cf.gov/about-us/blog for whatever reason, I get back staging.cf.gov/about-us/blog. If yes, +1